doc/wg/core/notes/core-notes-2021-01-15.md

```diff
@@ -0,0 +1,227 @@
+  might want to couple them is a compelling reason.
+- Leon: Think of exit as yield-forever
+- Brad: I think it would be difficult to explain to someone used to python exit
+  or whatever
+- Phil: Those are runtimes! Who knows what system calls they use.
+- Brad: Okay we can wrap things in abstraction I guess. Don't feel super
+  strongly on this.
+- Hudson: A little weird that yield is the system call to run callbacks, except
+  in the case of yield-exit.
+- Phil: Sure, but yield is also always what you use to give up control to the
+  kernel.
+- Leon: Yeah makes sense to me.
+- Hudson: Okay cool.
+- Phil: conclusion is I need to go back and reread this old discussion about
+  adding a yield-reboot and then based on that detailed reading I will do
+  whatever I conclude we agreed on before
+
+## Initial stack pointer and Break
+- Johnathan: Clarification on my email: at the end of the email I had a
+  question and if anyone knows the answer lets start with that.
+- Phil: I am not sure I understand the email
+- Johnathan: I am hoping we can change how the kernel sets up the initial app
+  break and stack pointer so as to make the userspace initialization easier.
+  Also want to know how to support apps that require less than 3kB of memory.
+- Brad: Background for those who are not aware. OG Tock kernel did all setup
+  for the application (relocation and all that). We changed this around 1.0.
+  But we still needed the kernel to setup the stack pointer so we could keep our
+  promise of stability for apps compiled at 1.0.
+- Branden: Even if the app is setting itself up, it needs some valid stack
+  pointer, right?
+- Brad: Yes on ARM (hardware uses the stack), no on risc-v.
+- Brad: This being arch dependent makes me think this belongs in arch/. The
+  kernel doesn't need to know where the stack is. Then we could do this
+  differently for ARM and RISC-V. But this question remains -- what is the
+  contract between arch/ and the c-init code in userspace.
+- Johnathan: For system calls the kernel might still need a stack.
+- Branden: But should only need a small stack
+- Johnathan: How easy is it to determine just how much stack is needed
+- Brad: There is a fixed size, but it is not the size of StoredState (as you
+  asked in your email)
+- Brad: But it is still arch dependent! This may even depend on whether
+  floating point is enabled!
+- Alistair: It is very different (8 vs 26 registers)
+- Branden: Okay Brad's argument for arch specific makes a lot of sense. Unless
+  this is really hard because of how the kernel interacts with all this
+- Brad: I don't think it will be too hard, but might be harder to keep the
+  debug info about all that.
+- Leon: I like the idea of arch specific. But if we have an arch which does not
+  need these things setup I would say we should make the least promises to
+  userspace as is reasonable.
+- Johnathan: The trick is that making a system call on risc-v saves stuff to
```

Comment:
Line 138: there should be an "if" in here, i.e. "The tick is that if making a system call ..."